lib/charms/mysql/v0/s3_helpers.py

```diff
@@ -57,8 +57,20 @@ def upload_content_to_s3(content: str, content_path: str, s3_parameters: Dict) -
             aws_secret_access_key=s3_parameters["secret-key"],
             region_name=s3_parameters["region"] or None,
         )
-
-        s3 = session.resource("s3", endpoint_url=s3_parameters["endpoint"])
+        verif = True
+        ca_chain = s3_parameters["tls-ca-chain"]
+        if ca_chain:
+            ca = "\n".join([base64.b64decode(s).decode() for s in ca_chain])
+            ca_file = tempfile.NamedTemporaryFile()
+            ca_file.write(ca.encode())
+            ca_file.flush()
+            verif = ca_file.name
```

Comment:
@sabaini , please add a `finally` on this try-catch with a `ca_file.close()`, so it can be deleted after usage, e.g.:

```suggestion
            verif = ca_file.name
            s3 = session.resource(
            "s3",
            endpoint_url=s3_parameters["endpoint"],
            verify=verif,
        )

        bucket = s3.Bucket(s3_parameters["bucket"])

        with tempfile.NamedTemporaryFile() as temp_file:
            temp_file.write(content.encode("utf-8"))
            temp_file.flush()
            bucket.upload_file(temp_file.name, content_path)
    except Exception as e:
        logger.exception(
            f"Failed to upload content to S3 bucket={s3_parameters['bucket']}, path={content_path}",
            exc_info=e,
        )
        return False
    finally:
        ca_file.close()

```

Reply:
Right, when `open` is used as a context manager w/ the `with` statement, it will implicitly be closed